sklearn/manifold/laplacian_eigenmap_.py

```diff
@@ -0,0 +1,495 @@
+            # in arpack fails. We fallback to lobpcg
+            eigen_solver = "lobpcg"
+            # Revert the laplacian to its opposite to have lobpcg work
+            laplacian *= -1
+
+    if eigen_solver == 'amg':
+        # Use AMG to get a preconditioner and speed up the eigenvalue
+        # problem.
+        if not sparse.issparse(laplacian):
+            warnings.warn("AMG works better for sparse matrices")
+        # lobpcg needs double precision floats
+        laplacian = check_array(laplacian, dtype=np.float64,
+                                accept_sparse=True)
+        laplacian = _set_diag(laplacian, 1, norm_laplacian)
+        ml = smoothed_aggregation_solver(check_array(laplacian, 'csr'))
+        M = ml.aspreconditioner()
+        X = random_state.rand(laplacian.shape[0], n_components + 1)
+        X[:, 0] = dd.ravel()
+        lambdas, diffusion_map = lobpcg(laplacian, X, M=M, tol=1.e-12,
+                                        largest=False)
+        embedding = diffusion_map.T * dd
+        if embedding.shape[0] == 1:
+            raise ValueError
+
+    elif eigen_solver == "lobpcg":
+        # lobpcg needs double precision floats
+        laplacian = check_array(laplacian, dtype=np.float64,
+                                accept_sparse=True)
+        if n_nodes < 5 * n_components + 1:
+            # see note above under arpack why lobpcg has problems with small
+            # number of nodes
+            # lobpcg will fallback to eigh, so we short circuit it
+            if sparse.isspmatrix(laplacian):
+                laplacian = laplacian.toarray()
+            lambdas, diffusion_map = eigh(laplacian)
+            embedding = diffusion_map.T[:n_components] * dd
+        else:
+            laplacian = _set_diag(laplacian, 1, norm_laplacian)
+            # We increase the number of eigenvectors requested, as lobpcg
+            # doesn't behave well in low dimension
+            X = random_state.rand(laplacian.shape[0], n_components + 1)
+            X[:, 0] = dd.ravel()
+            lambdas, diffusion_map = lobpcg(laplacian, X, tol=1e-15,
+                                            largest=False, maxiter=2000)
+            embedding = diffusion_map.T[:n_components] * dd
+            if embedding.shape[0] == 1:
+                raise ValueError
+
+    embedding = _deterministic_vector_sign_flip(embedding)
+    if drop_first:
+        return embedding[1:n_components].T
+    else:
+        return embedding[:n_components].T
+
+
+
+class LaplacianEigenmap(BaseEstimator):
+    """Laplacian Eigenmap embedding for non-linear dimensionality reduction.
+
+    Forms an affinity matrix given by the specified function and
+    applies spectral decomposition to the corresponding graph laplacian.
+    The resulting transformation is given by the value of the
+    eigenvectors for each data point.
+
+    Read more in the :ref:`User Guide <laplacian_eigenmaps>`.
```

Comment:
`laplacian_eigenmaps` --> `laplacian_eigenmap` (to match with the tag declared earlier...)